x-pack/platform/plugins/shared/task_manager/server/task_store.ts

```diff
@@ -313,16 +316,21 @@ export class TaskStore {
       const id = taskInstance.id || v4();
       const validatedTaskInstance =
         this.taskValidator.getValidatedTaskInstanceForUpdating(taskInstance);
+
       savedObject = await soClient.create<SerializedConcreteTaskInstance>(
         'task',
         {
           ...taskInstanceToAttributes(validatedTaskInstance, id),
           ...(userScope ? { userScope } : {}),
           ...(apiKey ? { apiKey } : {}),
+          runAt: getFirstRunAt({ taskInstance: validatedTaskInstance, logger: this.logger }),
```

Comment:
nit: should we move this into the `taskInstanceToAttributes` function?

Reply:
I tried it but it is used by `update` and `bulk_update` as well. That was why I asked if we should recalculate the runAt when schedule is updated but it turned into a bigger issue, we better handle it with a different investigation issue.